src/lib/src/service/persistence/persistence.service.ts

```diff
@@ -29,12 +37,31 @@ export class PersistenceService {
     return this.getHttpClient().get<WorkspaceElement>(this.listFilesUrl).toPromise();
   }
 
-  createResource(path: string, type: string): Promise<string> {
-    return this.getHttpClient().post(this.getURL(path), '', { responseType: 'text', params: { type: type } }).toPromise();
+  createResource(path: string, type: string): Observable<string | Conflict> {
+    return this.getHttpClient().post(this.getURL(path), '', {
+      observe: 'response',
+      responseType: 'text',
+      params: { type: type }
+    }).map(response => response.body).catch(response => {
+      if (response.status === HTTP_STATUS_CONFLICT) {
+        return Observable.of(new Conflict(response.error));
+      } else {
+        Observable.throw(new Error(response.error));
+      }
+    });
   }
 
-  deleteResource(path: string): Promise<string> {
-    return this.getHttpClient().delete(this.getURL(path), {responseType: 'text'}).toPromise();
+  deleteResource(path: string): Observable<string | Conflict> {
+    return this.getHttpClient().delete(this.getURL(path),  {
+      observe: 'response',
+      responseType: 'text',
+      }).map(response => response.body).catch(response => {
```

Comment:
I'd prefer to have .catch in a new line (in order to separate it from the mapping).